### ISSUE TITLE
PLW0120 (else clause on loop without a break statement) incorrectly flagged

### ISSUE THREAD

**Issue by eddyg** [charliermarsh/ruff#1136]
```shell
❯ ruff --version
ruff 0.0.169
```

Trying to understand why PLW0120 _("Else clause on loop without a break statement, remove the else and de-indent all the code inside it")_ is being raised.

Here's a simple example which causes PLW0120, even though it _does_ have a `break`:

```python
for page in range(1, 10):
    if page < 5:
        print(page)
    else:
        print("error")
        break
else:
    print("OK")
```

When the `break` is moved inside the `if`, PLW0120 goes away.

**Comment by charliermarsh** [comment]
It's _probably_ just missing the nested detection somehow.

**Comment by charliermarsh** [comment]
(But, definitely looks like a bug.)

**Comment by charliermarsh** [comment]
(Will take a look at this today.)

**Comment by charliermarsh** [comment]
Fixed in #1143.

I think the second example you included is actually correct (or, at least, consistent with Pylint), since if the loop exits early with a `return`, you'll never hit the `else` block code anyway -- so, IIUC, it's identical to removing the `else`.
